core/workspace_comment_svg.js

```diff
@@ -0,0 +1,591 @@
+ * @return {Element} The root SVG node (probably a group).
+ * @package
+ */
+Blockly.WorkspaceCommentSvg.prototype.getSvgRoot = function() {
+  return this.svgGroup_;
+};
+
+/**
+ * Returns this comment's text.
+ * @return {string} Comment text.
+ * @package
+ */
+Blockly.WorkspaceCommentSvg.prototype.getContent = function() {
+  return this.textarea_ ? this.textarea_.value : this.content_;
+};
+
+/**
+ * Set this comment's content.
+ * @param {string} content Comment content.
+ * @package
+ */
+Blockly.WorkspaceCommentSvg.prototype.setContent = function(content) {
+  Blockly.WorkspaceCommentSvg.superClass_.setContent.call(this, content);
+  if (this.textarea_) {
+    this.textarea_.value = content;
+  }
+};
+
+/**
+ * Update the cursor over this comment by adding or removing a class.
+ * @param {boolean} enable True if the delete cursor should be shown, false
+ *     otherwise.
+ * @package
+ */
+Blockly.WorkspaceCommentSvg.prototype.setDeleteStyle = function(enable) {
+  if (enable) {
+    Blockly.utils.addClass(
+        /** @type {!Element} */ (this.svgGroup_), 'blocklyDraggingDelete');
+  } else {
+    Blockly.utils.removeClass(
+        /** @type {!Element} */ (this.svgGroup_), 'blocklyDraggingDelete');
+  }
+};
+
+Blockly.WorkspaceCommentSvg.prototype.setAutoLayout = function() {
+  // NOP for compatibility with the bubble dragger.
+};
+
+/**
+ * Decode an XML comment tag and create a rendered comment on the workspace.
+ * @param {!Element} xmlComment XML comment element.
+ * @param {!Blockly.Workspace} workspace The workspace.
+ * @param {number=} opt_wsWidth The width of the workspace, which is used to
+ *     position comments correctly in RTL.
+ * @return {!Blockly.WorkspaceCommentSvg} The created workspace comment.
+ * @package
+ */
+Blockly.WorkspaceCommentSvg.fromXml = function(xmlComment, workspace,
+    opt_wsWidth) {
+  Blockly.Events.disable();
+  try {
+    var info = Blockly.WorkspaceComment.parseAttributes(xmlComment);
+
+    var comment = new Blockly.WorkspaceCommentSvg(workspace,
+        info.content, info.h, info.w, info.id);
+    if (workspace.rendered) {
+      comment.initSvg();
+      comment.render(false);
+    }
+    // Position the comment correctly, taking into account the width of a
+    // rendered RTL workspace.
+    if (!isNaN(info.x) && !isNaN(info.y)) {
+      if (workspace.RTL) {
+        var wsWidth = opt_wsWidth || workspace.getWidth();
+        comment.moveBy(wsWidth - info.x, info.y);
+      } else {
+        comment.moveBy(info.x, info.y);
+      }
+    }
+  } finally {
+    Blockly.Events.enable();
+  }
+  Blockly.WorkspaceComment.fireCreateEvent(comment);
+
+  return comment;
+};
+
+/**
+ * Encode a comment subtree as XML with XY coordinates.
+ * @param {boolean=} opt_noId True if the encoder should skip the comment id.
+ * @return {!Element} Tree of XML elements.
+ * @package
+ */
+Blockly.WorkspaceCommentSvg.prototype.toXmlWithXY = function(opt_noId) {
```

Comment:
same question here about calling superclass version of this function first (or maybe even just in the LTR case)?